go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -1005,10 +1005,13 @@ func (bnc *BaseNetworkController) DeleteResourceCommon(objType reflect.Type, obj
 			return fmt.Errorf("could not cast obj of type %T to *knet.NetworkPolicy", obj)
 		}
 		netinfo, err := bnc.networkManager.GetActiveNetworkForNamespace(knp.Namespace)
-		if err != nil {
-			return fmt.Errorf("could not get active network for namespace %s: %v", knp.Namespace, err)
+		// The InvalidPrimaryNetworkError error is thrown when UDN is not found because
+		// it has been already deleted, so just proceed with deleting NetworkPolicy in
+		// such a scenario as well.
+		if err != nil && !util.IsInvalidPrimaryNetworkError(err) {
+			return fmt.Errorf("could not get active network for namespace %s: %w", knp.Namespace, err)
```

Comment:
nit do we also need the networkpolicy name for easier debugging ? we could have more than 1 netpol in this namespace and we won't know which one's deletion failed?

Reply:
I think it should be fine as is, the network is an attribute of the whole namespace so having a network policy name here is not strictly necessary

Reply:
`I think it should be fine as is, the network is an attribute of the whole namespace so having a network policy name here is not strictly necessary`

hmm what I mean is how do you which network policy's deletion here got screwed up?